File: backend/src/main/java/com/travelquest/dto/AdventurePublicDTO.java
package com.travelquest.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class AdventurePublicDTO {
    private Long id;
    private String name;
    private String location;
    private List<String> tags;
    private String description;
    private double rating;
    private String link;
    private double latitude;
    private double longitude;
    private List<String> imageUrls;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String author;
}

File: frontend/types/AdventurePublicDTO.ts
export interface AdventurePublicDTO {
  id: number;
  name: string;
  location: string;
  tags: string[];
  description: string;
  rating: number;
  link: string;
  latitude: number;
  longitude: number;
  imageUrls: string[];
  createdAt: string;
  author: string;
  likesCount: number;
  likedByCurrentUser: boolean;
}